src/lib/support/logging/CHIPLogging.cpp

```diff
@@ -199,6 +199,115 @@ DLL_EXPORT void SetLogFilter(uint8_t category)
 }
 #endif // CHIP_LOG_FILTERING
 
+/**
+ * This static method outputs a line of the memory dump.
+ *
+ * @param[in]  module    The LogModule generating the log message.
+ * @param[in]  category  The LogCategory for filtering log level.
+ * @param[in]  aBuf      A pointer to the buffer.
+ * @param[in]  aLength   Number of bytes in the buffer.
+ *
+ */
+static void DumpLine(uint8_t module, uint8_t category, const void * aBuf, const size_t aLength)
+{
+    char buf[80];
+    char * cur = buf;
+
+    snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), "|");
+    cur += strlen(cur);
+
+    for (size_t i = 0; i < 16; i++)
+    {
+        if (i < aLength)
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " %02X", ((uint8_t *) (aBuf))[i]);
+            cur += strlen(cur);
+        }
+        else
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " ..");
+            cur += strlen(cur);
+        }
+
+        if (!((i + 1) % 8))
+        {
+            snprintf(cur, sizeof(buf) - static_cast<size_t>(cur - buf), " |");
+            cur += strlen(cur);
```

Comment:
For many of these `cur += strlen(cur)`, they could be replaced by constant lengths, rather than needing a call to strlen.